src/TLuaInterpreter.cpp

```diff
@@ -6363,27 +6364,22 @@ int TLuaInterpreter::getMudletHomeDir(lua_State* L)
 // Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#getMudletLuaDefaultPaths
 int TLuaInterpreter::getMudletLuaDefaultPaths(lua_State* L)
 {
-    int index = 1;
+    int index = 0;
+    // This is a static method so does not have a this pointer - so go and get
+    // the one used by the profile so we can read the instance's data:
+    auto& host = getHostFromLua(L);
+    auto* pLua = host.getLuaInterpreter();
+    Q_ASSERT_X(pLua, "TLuaInterpreter::getMudletLuaDefaultPaths", "nullptr received when looking for the instantiated instance of TLuaInterpreter for a Host.");
+
+    QStringListIterator itPath(pLua->mPossiblePaths);
     lua_newtable(L);
-#if defined(Q_OS_MACOS)
-    lua_createtable(L, 3, 0);
-#else
-    lua_createtable(L, 2, 0);
-#endif
-    // add filepath relative to the binary itself (one usecase is AppImage on Linux)
-    QString nativePath = QDir::toNativeSeparators(QCoreApplication::applicationDirPath() + "/mudlet-lua/lua/");
-    lua_pushstring(L, nativePath.toUtf8().constData());
-    lua_rawseti(L, -2, index++);
-#if defined(Q_OS_MACOS)
-    // add macOS lua path relative to the binary itself, which is part of the Mudlet.app package
-    nativePath = QDir::toNativeSeparators(QCoreApplication::applicationDirPath() + "/../Resources/mudlet-lua/lua/");
-    lua_pushstring(L, nativePath.toUtf8().constData());
-    lua_rawseti(L, -2, index++);
-#endif
-    // add the default search path as specified by build file
-    nativePath = QDir::toNativeSeparators(LUA_DEFAULT_PATH "/");
-    lua_pushstring(L, nativePath.toUtf8().constData());
-    lua_rawseti(L, -2, index++);
+    while (itPath.hasNext()) {
```

Comment:
Could you use a modern `for` loop here?

Reply:
:thinking: I suppose so, but I would need an index for the Lua table and that would prefer the use of a *traditional* index variable `for` loop to provide the index for the Lua table but it would be one more than the index for the `QStringList::at(int)` so there would need to be a `+1` adjustment, or the increment to be done inside the loop. TBH This seems to me to be a perfectly adequate implementation that does exactly what is needed...

Reply:
I agree, I missed the need for an index.